components/Habits/HabitDailyView.tsx

```diff
@@ -0,0 +1,89 @@
+import { css } from '@emotion/react';
+import { useTranslation } from 'next-i18next';
+import Image from 'next/image';
+import { HabitWithHistory } from '../../src/types/habits';
+import CompleteButton from '../entries/CompleteButton';
+import calculateCompletionPercentages from './utils';
+
+const styles = {
+  content: css({
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    backgroundColor: '#333',
+    margin: '12px 0px',
+    padding: '12px',
+    gap: '8px',
+  }),
+
+  middleContainer: css({
+    display: 'flex',
+    flexDirection: 'column',
+    textAlign: 'left',
+    minWidth: '30%',
+  }),
+  deadline: css({
+    color: '#91E220',
+  }),
+
+  streakContainer: css({
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    maxWidth: '100px',
+    padding: '0px 12px',
+  }),
+  streakText: css({
+    color: '#fad15c',
+    paddingLeft: '4px',
+  }),
+};
+
+type HabitDailyViewProps = {
+  habitWithHistory: HabitWithHistory;
+  streak: number;
+  date: string;
+};
+
+const HabitDailyView = ({
+  habitWithHistory,
+  streak,
+  date,
+}: HabitDailyViewProps) => {
+  const { t } = useTranslation(['common', 'habit']);
+  const completionPercentagesByDay = calculateCompletionPercentages({
+    habitWithHistory,
+    dates: [date],
+  });
+
+  return (
+    <div css={styles.content}>
+      <div css={styles.streakContainer}>
+        <Image
+          src='/images/fire.svg'
+          alt={t('habit:alt.fire')}
+          width='32'
+          height='32'
+        />
+        <span css={styles.streakText}>{streak}</span>
+      </div>
+
+      <div css={styles.middleContainer}>
+        <span>{habitWithHistory.name}</span>
+        <span css={styles.deadline}>
+          {completionPercentagesByDay[date] < 100
+            ? t('common:dayLeft', { quantity: '2' })
+            : t('habit:complete')}
+        </span>
+      </div>
+
+      {habitWithHistory.target.quantity === 1 ? (
```

Comment:
```
      {habitWithHistory.target.quantity === 1 ? (
```

is the sort of condition that could do with a variable backing it for readability, or a function for re-use. That way, you can indicate intent of what, rather than how here (I myself am not 100% on what this is, but I recall it relating to binary habit, so perhaps `isBinaryHabit(habitWithHistory)`

Same comment around the `completionPercentagesByDay[date] < 100`, could become `isComplete()` or similar